heat/core/statistics.py

```diff
@@ -533,6 +536,139 @@ def local_min(*args, **kwargs):
+            [ 0.2857,  0.6898, -0.6331,  0.8795],
+            [-0.6842,  0.4533,  0.2912, -0.8317]])
+
+    >>> ht.minimum(a,b)
+    tensor([[-0.1955, -0.9656, -0.8029,  0.2366],
+            [-0.4212, -0.5107, -1.5727, -0.1232],
+            [-0.6842, -1.8313,  0.2912, -1.2770]])
+
+    >>> c = ht.random.randn(1,4)
+    >>> c
+    tensor([[-1.6428,  0.9803, -0.0421, -0.8206]])
+
+    >>> ht.minimum(a,c)
+    tensor([[-1.6428, -0.9656, -0.0421, -0.8206],
+            [-1.6428, -0.5107, -1.5727, -0.8206],
+            [-1.6428, -1.8313, -0.0421, -1.2770]])
+
+    >>> b.__setitem__((0,1), ht.nan) 
+    >>> b
+    tensor([[ 0.8310,     nan, -0.8029,  0.2366],
+            [ 0.2857,  0.6898, -0.6331,  0.8795],
+            [-0.6842,  0.4533,  0.2912, -0.8317]])
+    >>> ht.minimum(a,b)
+    tensor([[-0.1955,     nan, -0.8029,  0.2366],
+            [-0.4212, -0.5107, -1.5727, -0.1232],
+            [-0.6842, -1.8313,  0.2912, -1.2770]])
+
+    >>> d = ht.random.randn(3,4,5)
+    >>> ht.minimum(a,d)
+    ValueError: operands could not be broadcast, input shapes (3, 4) (3, 4, 5)
+    '''
+    # perform sanitation
+    if not isinstance(x1, dndarray.DNDarray) or not isinstance(x2, dndarray.DNDarray):
+        raise TypeError('expected x1 and x2 to be a ht.DNDarray, but were {}, {} '.format(type(x1), type(x2)))
+    if out is not None and not isinstance(out, dndarray.DNDarray):
+        raise TypeError('expected out to be None or an ht.DNDarray, but was {}'.format(type(out)))
+
+    # apply split semantics
+    if x1.split is not None or x2.split is not None:
+        if x1.split == None:
+            x1.resplit(x2.split)
+        if x2.split == None:
+            x2.resplit(x1.split)
+        if x1.split != x2.split:
+            if np.prod(x1.gshape) < np.prod(x2.gshape):
+                x1.resplit(x2.split)
+            if np.prod(x2.gshape) < np.prod(x1.gshape):
+                x2.resplit(x1.split)
+            else:
+                if x1.split < x2.split:
+                    x2.resplit(x1.split)
+                else:
+                    x1.resplit(x2.split)
+    split = x1.split
+
+    # locally: apply torch.min(x1, x2)
+    output_lshape = stride_tricks.broadcast_shape(x1.lshape, x2.lshape)
+    lresult = factories.empty(output_lshape)
+    lresult._DNDarray__array = torch.min(x1._DNDarray__array, x2._DNDarray__array)
+    lresult._DNDarray__dtype = types.promote_types(x1.dtype, x2.dtype)
+    lresult._DNDarray__split = split
+    if x1.split is not None or x2.split is not None:
+        if x1.comm.is_distributed():  # assuming x1.comm = x2.comm
+            output_gshape = stride_tricks.broadcast_shape(x1.gshape, x2.gshape)
+            result = factories.empty(output_gshape)
+            x1.comm.Allgather(lresult, result)
+            # TODO: adopt Allgatherv() as soon as it is fixed, Issue #233
+            result._DNDarray__dtype = lresult._DNDarray__dtype
+            result._DNDarray__split = split
+
+            if out is not None:
```

Comment:
is it possible to combine this `if out` and the other one?

Reply:
I'm not sure how, as one of them only writes out in the distributed case, the other one in the non-distributed case. If you can think of something, let me know.